src/backend/app/auth/auth_routes.py

```diff
@@ -81,28 +91,21 @@ async def callback(request: Request, osm_auth=Depends(init_osm_auth)):
     # Get access token
     access_token = osm_auth.callback(callback_url).get("access_token")
     log.debug(f"Access token returned of length {len(access_token)}")
-    response = Response(status_code=HTTPStatus.OK)
 
-    # Set cookie
-    cookie_name = settings.FMTM_DOMAIN.replace(".", "_")
-    log.debug(
-        f"Setting cookie in response named '{cookie_name}' with params: "
-        f"max_age=31536000 | expires=31536000 | path='/' | "
-        f"domain={settings.FMTM_DOMAIN} | httponly=True | samesite='lax' | "
-        f"secure={False if settings.DEBUG else True}"
-    )
-    response.set_cookie(
-        key=cookie_name,
-        value=access_token,
-        max_age=31536000,  # OSM currently has no expiry
-        expires=31536000,  # OSM currently has no expiry
-        path="/",
-        domain=settings.FMTM_DOMAIN,
-        secure=False if settings.DEBUG else True,
-        httponly=True,
-        samesite="lax",
-    )
-    return response
+    osm_user = osm_auth.deserialize_access_token(access_token)
```

Comment:
Do we need error handling here of the OSM token fails to deserialise? That would essentially mean OSM login failed and we should not proceed

Reply:
We are deserializing to get the user data from osm and encode it again using jwt. What do you mean by proceeding after osm login failed? How should we proceed even after osm login fails? If the user fails osm login, they can choose a temporary login.

Reply:
If the token fails to deserialise then we get two scenarios:
- the JWT is generated with empty OSM details
- a possible 500 response

Neither scenario is great. We should handle failure gracefully and return a HTTPError